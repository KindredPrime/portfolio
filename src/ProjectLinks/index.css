.ProjectLinks {
  text-align: center;
  margin-top: 0px;
  margin-bottom: 25px;
}

.ProjectLinks__divider {
  margin-left: 8px;
  margin-right: 8px;
  font-weight: 900;
  font-size: larger;
}