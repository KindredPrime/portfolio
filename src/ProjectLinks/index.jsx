import PropTypes from 'prop-types';
import './index.css';

function ProjectLinks(props) {
  const { repo, live } = props;

  return (
    <div className="ProjectLinks">
      <a
        href={repo}
        target="_blank"
        rel="noreferrer"
        className="ProjectLinks__repo"
      >
        GitHub
      </a>
      <span className="ProjectLinks__divider">|</span>
      <a
        href={live}
        target="_blank"
        rel="noreferrer"
        className="ProjectLinks__live"
      >
        Live
      </a>
    </div>
  );
}

ProjectLinks.propTypes = {
  repo: PropTypes.string.isRequired,
  live: PropTypes.string.isRequired
}

export default ProjectLinks;