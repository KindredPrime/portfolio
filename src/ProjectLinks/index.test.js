import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import ProjectLinks from './index';

describe('ProjectLinks Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ProjectLinks repo="" live="" />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders content as expected', () => {
    render(
      <ProjectLinks
        repo="https://www.repo.com"
        live="https://www.live.com"
      />
    );

    expect(document.body).toMatchSnapshot();
  });
});