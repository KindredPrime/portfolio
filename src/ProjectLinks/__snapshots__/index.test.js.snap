// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectLinks Component renders content as expected 1`] = `
<body>
  <div>
    <div
      class="ProjectLinks"
    >
      <a
        class="ProjectLinks__repo"
        href="https://www.repo.com"
        rel="noreferrer"
        target="_blank"
      >
        GitHub
      </a>
      <span
        class="ProjectLinks__divider"
      >
        |
      </span>
      <a
        class="ProjectLinks__live"
        href="https://www.live.com"
        rel="noreferrer"
        target="_blank"
      >
        Live
      </a>
    </div>
  </div>
</body>
`;
